test/constructorio-synonyms-one-way.js

```diff
@@ -0,0 +1,587 @@
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+
+    it('should return no results when supplying an invalid num results per page and page combination', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getOneWaySynonyms({
+        num_results_per_page: 1,
+        page: 7,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response).to.have.property('one_way_synonym_relations').an('array').length(0);
+        done();
+      });
+    });
+
+    it('should return an error when supplying an invalid page parameter', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getOneWaySynonyms({
+        page: 'abc',
+      }, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message', 'page must be an integer');
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+
+    it('should return an error when supplying an invalid key/token', (done) => {
+      const constructorio = new Constructorio({
+        apiToken: 'bad-token',
+        apiKey: 'bad-key',
+      });
+
+      constructorio.getOneWaySynonyms({}, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message').to.match(/You have supplied an invalid/);
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+  });
+
+  describe('getOneWaySynonym', () => {
+    const oneWaySynonymPhrase = 'condiments';
+
+    before((done) => {
+      // Create test one way synonyms for use in tests
+      addTestOneWaySynonym(oneWaySynonymPhrase).then(done).catch(done);
+    });
+
+    after((done) => {
+      // Clean up test one way synonyms
+      removeTestOneWaySynonym(oneWaySynonymPhrase).then(done).catch(done);
+    });
+
+    it('should retrieve a list of one way synonyms when supplying a valid phrase', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getOneWaySynonym({
+        phrase: oneWaySynonymPhrase,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response).to.have.property('one_way_synonym_relations').an('array').length(1);
+        done();
+      });
+    });
+
+    it('should return no results when supplying a non-existent phrase', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getOneWaySynonym({
+        phrase: 1,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response).to.have.property('one_way_synonym_relations').an('array').length(0);
+        done();
+      });
+    });
+
+    it('should return no results when not supplying a phrase', (done) => {
```

Comment:
Surprised this doesn't throw an error!

Reply:
I added a check on the client side since it ends up treating `"undefined"` as the phrase if it isn't provided, which is not right... So it will now throw an error if a phrase isn't provided.